cowbird/handlers/impl/magpie.py

```diff
@@ -143,10 +271,88 @@ def create_permission(self, permissions_data):
             if resp.status_code == 200:
                 LOGGER.info("Permission creation was successful.")
             else:
-                raise HTTPError(f"Failed to create permissions : {resp.text}")
+                raise HTTPError(f"HTTPError {resp.status_code} - Failed to create permissions : {resp.text}")
         else:
             LOGGER.warning("Empty permission data, no permissions to create.")
 
+    def create_permission_by_res_id(self, res_id, perm_name, perm_access, perm_scope, user_name="", grp_name=""):
+        # type: (int, str, str, str, Optional[str], Optional[str]) -> None
+
+        if user_name:
+            url = f"{self.url}/users/{user_name}/resources/{res_id}/permissions"
+        elif grp_name:
+            url = f"{self.url}/groups/{grp_name}/resources/{res_id}/permissions"
+        else:
+            raise ValueError("Trying to create a permission, but missing an input user name or group name.")
+
+        resp = self._send_request(method="GET", url=url)
+        if resp.status_code != 200:
+            raise HTTPError(f"HTTPError {resp.status_code} - Failed to find resource: {resp.text}")
+
+        # By default, POST to create a new permission, but check before if the permission already exists, to avoid
+        # unnecessary events in Magpie.
+        method = "POST"
+        for perm in resp.json()["permissions"]:
+            if perm["name"] == perm_name:
+                if perm["access"] == perm_access and perm["scope"] == perm_scope:
```

Comment:
Valid use case not covered by test (as well as few other lines below, e.g. `PUT` method case). 
Add it if it can be done quickly (copy-paste some existing test/setup).
Otherwise, can be ignored.